mobile/src/components/course/course.tsx

```diff
@@ -38,11 +40,29 @@ const Course: FC = () => {
     });
   }, []);
 
-  useEffect(() => {
-    if (course) {
-      dispatch(courseModulesActions.getCourseModules({ courseId: course.id }));
-    }
-  }, [course]);
+  useFocusEffect(
+    useCallback(() => {
+      if (course) {
+        dispatch(
+          courseModulesActions.getCourseModules({ courseId: course.id }),
+        );
+        dispatch(
+          coursesActions.getMentorsByCourseId({
+            courseId: course.id,
+            filteringOpts: {
+              mentorName: '',
+            },
+          }),
+        )
+          .unwrap()
+          .then(() => dispatch(coursesActions.updateVisibilityBecomeMentor()));
```

Comment:
we only use this when we can't do some actions inside asyncThunk, for example navigate to another screen. In this case we can dispatch action in asyncThunk